Add federated services docs (#1874)
linkerd/website: linkerd.io/content/2-edge/reference/multicluster.md

Linkerd's [multi-cluster functionality](../../features/multicluster/) allows
pods to connect to Kubernetes services across cluster boundaries in a way that
is secure and fully transparent to the application. This feature supports three

> modes: hierarchical (using an gateway), flat (without a gateway), and federated.

Reviewer: ```suggestion
modes: hierarchical (using a gateway), flat (without a gateway), and federated.
```

* **Flat mode** requires that all pods on the source cluster be able to directly
  connect to pods on the destination cluster.
* **Hierarchical mode** only requires that the gateway IP of the destination

>   cluster be reachable by pods on the source cluster.

Reviewer: I realize you didn't change this in this PR, but can we swap these two modes so the order of the descriptions are heirarchical, flat, federated to match the "three modes" text above?

a kubeconfig file that allows an API client to be configured. The kubeconfig
file uses RBAC to provide the "principle of least privilege", ensuring the
*destination service* may only access only the resources it needs.

## Federated Services

Federated services take this a step farther by allowing a service which is

> deployed to multiple clusters to be joined into a single union service.

Reviewer: ```suggestion
deployed to multiple clusters to be joined into a single unified service.
```

maybe?

> deployed to multiple clusters to be joined into a single federated service.

Reviewer 2: Maybe single "logical" service to avoid a recursive definition?

Reviewer 2: I think I saw "union" elsewhere, that might be an option here too.

The service mirror controller will look for all services in all linked clusters
which match a label selector (`mirror.linkerd.io/federated=member` by default)
and create a federated service called `<svc name>-federated` which will act as
a union of all those services with that name. For example, all traffic sent to

> the `store-web-federated` federated service will be load balananced over all

Reviewer 2: ```suggestion
the `store-web-federated` federated service will be load balanced over all
```

> replicas of all services name `store-web` in all linked clusters.

Reviewer 3: ```suggestion
replicas of all services named `store-web` in all linked clusters.
```

The concept of "namespace sameness" applies, which means that the federated
service will be created in the same namespace as the individual services and
services can only join a federated service in the same namespace.

> Since Linkerd's *destination servce* uses "remote-discovery" to discovery the

Reviewer 2: ```suggestion
Since Linkerd's *destination service* uses "remote-discovery" to discover the
```
